rbs/AssertionsRewriter.cc

```diff
@@ -484,16 +484,10 @@ unique_ptr<parser::Node> AssertionsRewriter::rewriteNode(unique_ptr<parser::Node
             send->receiver = maybeInsertCast(move(send->receiver));
             send->receiver = rewriteNode(move(send->receiver));
 
-            if (send->method == core::Names::squareBracketsEq()) {
-                // This is a `foo[key]=(y)` method, walk y for chained method calls
+            if (send->method == core::Names::squareBracketsEq() || send->method.isSetter(ctx.state)) {
                 send->args.back() = rewriteNode(move(send->args.back()));
                 result = move(node);
                 return;
-            } else if (send->method.isSetter(ctx.state) && send->args.size() == 1) {
-                // This is a `foo.x=(y)` method, we treat it as a `x = y` assignment
-                send->args[0] = maybeInsertCast(move(send->args[0]));
-                result = move(node);
-                return;
```

Comment:
It looks like `[]=` can take multiple parameters:

```ruby
class A
  def []=(x, y, value)
  end
end

A.new[1, 2] = 3
```

So you may want to fix that in a followup?  I think it would make sense to rewrite all args of `[]=` all the time?